### PR TITLE
CI docbuild: Do not rebuild sagelib from scratch

### PR DESCRIPTION
**Pull request by mkoeppe**
Reworking #35652 as discussed in https://github.com/sagemath/sage/pull/36469#issuecomment-1765662418

Also applying the changes made in #35866 (cosmetic improvements to the Actions logs of doc-build.yml) to doc-build-pdf.yml

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/doc-build.yml [hunk 1]
```diff
@@ -124,7 +124,7 @@ jobs:
           set -ex
           export SAGE_USE_CDNS=yes
           mv /sage/local/share/doc/sage/html/en/.git /sage/.git-doc
-          make doc-clean doc-uninstall sagelib-clean && git clean -fx src/sage
+          make doc-clean doc-uninstall
           mkdir -p /sage/local/share/doc/sage/html/en/ && mv /sage/.git-doc /sage/local/share/doc/sage/html/en/.git
           ./config.status && make doc-html
```
Comment (kwankyu) [inline .github/workflows/doc-build.yml:129]: Not relevant here, but I wonder what `/sage/local/share/doc/sage/html/en/.git` is, and why it is important to preserve this across `doc-clean doc-uninstall`.

Comment (mkoeppe) [reply .github/workflows/doc-build.yml:129]: This is how we keep track of the changes in the built documentation, compared to what is contained in the Docker image.

Comment (kwankyu) [reply .github/workflows/doc-build.yml:129]: Ah, I see. You use this information to create "changes" file list (as seen below) of the doc.

Comment (kwankyu) [reply .github/workflows/doc-build.yml:129]: By the way, the "changes" list seems not working properly...

**Comment by kwankyu** [comment]
Now the doc build is very fast. Very nice.

**Review by kwankyu — APPROVED** [review]
LGTM.

**Comment by kwankyu** [comment]
Thanks to Tobias too!

**Comment by tobiasdiez** [comment]
Currently `make doc` reinstalls sage_conf and sage_lib. Would it be easy to extract this to a new job to make the output easier to understand.

**Comment by tobiasdiez** [comment]
Another suggestion for improvement: use `continue-on-error: true` on the incremental step. Then a failure of this step doesn't cause the whole job/workflow to fail.

File: .github/workflows/doc-build.yml (hunk 1 above)
Comment (mkoeppe) [reply .github/workflows/doc-build.yml:129]: > the "changes" list seems not working properly...

This is from a recent change to CDN use for mathjax -- see the diff: https://deploy-preview-36473--sagemath-tobias.netlify.app/html.diff

Comment (mkoeppe) [reply .github/workflows/doc-build.yml:129]: I've opened https://github.com/sagemath/sage/pull/36475 for fixing this

**Comment by mkoeppe** [comment]
> Another suggestion for improvement: use `continue-on-error: true` on the incremental step. Then a failure of this step doesn't cause the whole job/workflow to fail.

Good idea, but obsolete if #36446 works well enough.

**Pull request merged**